### PR TITLE
Bind faxes to many users

### PR DESCRIPTION
**Pull request by danigargar**
Added new feature that enables a client administrator to assign a fax to multiple users

#### Type Of Change 
- [ ] Small bug fix
- [x] New feature or enhancement
- [ ] Breaking change

#### Checklist:
- [x] Commits are named and have tag following [commit rules](https://github.com/irontec/ivozprovider/blob/bleeding/doc/dev/en/commits.md)
- [x] Commits are split per component (schema, web/admin, kamusers, agis, ..)
- [x] Changes have been tested locally
- [ ] Fixes an existing issue (Fixes #XXXX) 
- [ ] Upport from existing Pull request #XXXX

### CODE REVIEW

**Review by ironArt3mis — CHANGES_REQUESTED** [review]
Functional review required

File: library/Ivoz/Provider/Domain/Model/Fax/FaxDto.php [hunk 1]
```diff
@@ -2,8 +2,27 @@
...
+    public const CONTEXTS_WITH_REL_USERS = [
+        self::CONTEXT_WITH_REL_USERS,
+        self::CONTEXT_DETAILED
+    ];
+
+   /** @var int[]
+    * @AttributeDefinition  (
+    *     type="array",
+    *     collectionValueType="int",
+    *     description="Voicemail rel users"
```
Comment (mmadariaga) [inline library/Ivoz/Provider/Domain/Model/Fax/FaxDto.php:21]:
```suggestion
    *     description="Fax rel users"
```

File: library/Ivoz/Provider/Domain/Model/Fax/FaxDto.php [hunk 2]
```diff
@@ -41,4 +64,39 @@ public function denormalize(array $data, string $context, string $role = ''): vo
             $data
         );
     }
+
+    /**
+     * @return array<array-key, mixed>
+     */
+    public function normalize(string $context, string $role = ''): array
+    {
+
```
Comment (mmadariaga) [inline library/Ivoz/Provider/Domain/Model/Fax/FaxDto.php:73]: Remove this line

File: library/Ivoz/Provider/Domain/Model/Fax/FaxDto.php [hunk 3]
```diff
@@ -41,4 +64,39 @@ public function denormalize(array $data, string $context, string $role = ''): vo
...
+     * @return void
+     */
+    public function setRelUserIds(array $userIds): void
+    {
+        $this->relUserIds = $userIds;
+
+        $faxesRelUsers = [];
+        foreach ($userIds as $id) {
+            $dto = new FaxesRelUserDto();
+            $dto->setUserId($id);
```
Comment (mmadariaga) [inline library/Ivoz/Provider/Domain/Model/Fax/FaxDto.php:95]:
```suggestion
            $dto = new FaxesRelUserDto($id);
```
just if you prefer, no need to change it

**Review by mmadariaga — CHANGES_REQUESTED** [review]

File: library/Ivoz/Provider/Domain/Model/Fax/FaxDto.php (hunk 3 above)
Comment (danigargar) [reply library/Ivoz/Provider/Domain/Model/Fax/FaxDto.php:95]: Not the same, but I notice $id should be $userId.

**Review by mmadariaga — APPROVED** [review]

**Review by ironArt3mis — APPROVED** [review]

**Review by ironArt3mis — APPROVED** [review]

**Pull request merged**